NavigateU/NavigateU/EditProfileScreen/EditProfileViewController.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  EditProfileViewController.swift
+//  NavigateU
+//
+//  Created by Faki Doosuur Doris on 15.08.2024.
+//
+
+import UIKit
+
+class EditProfileViewController: UIViewController, EditProfileDelegate {
+
+    var editProfileView: EditProfileView?
+    let viewModel: EditProfileViewModel
+
+    init(viewModel:EditProfileViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+        viewModel.fetchUserProfile()
+        viewModel.onProfileDetails = { [weak self] in
+            self?.loadPage()
+        }
+    }
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        viewModel.fetchUserProfile()
+    }
+
+
+    func setupView() {
+        editProfileView = EditProfileView(frame: view.bounds)
+        view = editProfileView
+        editProfileView?.delegate = self
+        editProfileView?.backgroundColor = .white
+    }
+
+    private func loadPage() {
+        self.editProfileView?.firstName.text = self.viewModel.firstName
+        self.editProfileView?.lastName.text = self.viewModel.lastName
+        self.editProfileView?.email.text = self.viewModel.email
+        self.editProfileView?.password.text = self.viewModel.password
+        self.editProfileView?.cityOfResidence.text = self.viewModel.cityOfResidence
+        self.editProfileView?.countryOfOrigin.text = self.viewModel.nationality
+    }
```

Comment:
it's not how MVVM works. You better rename viewModel to presenter (in whole project, if everywhere is the same implementation), because the main thing in MVVM is subscription, not the direct call of properties